File: testng-core/src/test/java/test/hook/samples/ConfigurableFailureSample.java
package test.hook.samples;

import static test.hook.HookableTest.HOOK_INVOKED_ATTRIBUTE;

import org.testng.IConfigureCallBack;
import org.testng.ITestResult;
import org.testng.annotations.Test;

public class ConfigurableFailureSample extends BaseConfigurableSample {

  @Override
  public void run(IConfigureCallBack callBack, ITestResult testResult) {
    testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true");
    // Not calling the callback
  }

  @Test
  public void hookWasNotRun() {}
}

File: testng-core/src/test/kotlin/test/hook/samples/ConfigurableFailureSample.kt
package test.hook.samples

import org.testng.IConfigureCallBack
import org.testng.ITestResult
import org.testng.annotations.Test
import test.hook.HOOK_INVOKED_ATTRIBUTE

class ConfigurableFailureSample:BaseConfigurableSample() {

    override fun run(callBack: IConfigureCallBack?, testResult: ITestResult) {
        testResult.setAttribute(HOOK_INVOKED_ATTRIBUTE, "true")
    }
  @Test
  fun hookWasNotRun() {}
}
